scripts/train_linker.py

```diff
@@ -0,0 +1,81 @@
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.metrics import accuracy_score, classification_report
+from joblib import dump, load
+import jsonlines
+from statistics import mean
+import datetime
+from scispacy import umls_semantic_type_tree
+from linking import featurizer
+import argparse
+
+def main(data_path: str):
+    train = []
+    dev = []
+    test = []
+    root = ''
+    files = [f'{data_path}/train.jsonl', f'{data_path}/dev.jsonl', f'{data_path}/test.jsonl']
+    lists = [train, dev, test]
+    # limits = [40000000, 4000000, 4000000]
+    limits = [4000000, 0, 4000000]
+
+    start_time = datetime.datetime.now()
+
+    # load data from files
+    for filename, data, limit in zip(files, lists, limits):
+        with jsonlines.open(filename) as f:
+            for line in f:
+                data.append(line)
+                if len(data) % 100000 == 0:
+                    print(len(data))
+                if len(data) >= limit:
+                    break
+
+    end_time = datetime.datetime.now()
+    total_time = end_time - start_time
+    print(f'Time: {total_time.total_seconds()} seconds')
+
+    x = []
+    y = []
+    for data in lists:
+        features  = [featurizer(d) for d in data]
+        x.append(features)
+        y.append([d['label'] for d in data])
+
+    end_time = datetime.datetime.now()
+    total_time = end_time - start_time
+    print(f'Time: {total_time.total_seconds()} seconds')
+
+    cls = GradientBoostingClassifier(verbose=1)
```

Comment:
did you do any hyperparameter search? I think its usually pretty important for gradient boosting trees classifiers

Reply:
tried changing number of estimators and number of levels, no significant impact